lennonFolder/lennon-github-serach/simple-github/src/main/kotlin/com/androidhuman/example/simplegithub/api/GithubApiProvider.kt

```diff
@@ -0,0 +1,62 @@
+package com.androidhuman.example.simplegithub.api
+
+import android.content.Context
+import com.androidhuman.example.simplegithub.data.AuthTokenProvider
+import okhttp3.Interceptor
+import okhttp3.OkHttpClient
+import okhttp3.Response
+import okhttp3.logging.HttpLoggingInterceptor
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import java.io.IOException
+
+fun provideAuthApi(): AuthApi = Retrofit.Builder()
+        .baseUrl("https://github.com/")
```

Comment:
이 프로젝트에선 api를 2개밖에 사용 안하는데 보통 url를 const로 지정해주고 사용합니다.

이렇게 하는 이유엔 2가지 정도 이점이 있어요
- 추후 base url 변경시 수정이 편리하다.
- 내부적으로 .baseUrl()을 할때 마다 string 객체를 반복적으로 만드는것을 방지할 수 있다.
```kotlin
companion object {
const val BASE_GITHUB_URL = "https://github.com/"
const val BASE_API_GITHUB_URL = "https://api.github.com/"
}

Reply:
GithubApiProvider.kt은 클래스없이 top-level 함수로 정의되어 있는데 클래스 없이 companion object를 사용하는 방법이 있을까요?

Reply:
아 그렇군여 그렇담 const val 로만 지정해주면 됩니다~

```suggestion
const val BASE_GITHUB_URL = "https://github.com/"
const val BASE_API_GITHUB_URL = "https://api.github.com/"
```

Reply:
수정 완료!